indigo_content_api/v1/views.py

```diff
@@ -78,17 +78,31 @@ def parse_frbr_uri(self, frbr_uri):
         return frbr_uri
 
     def determine_place(self):
-        parts = self.kwargs['frbr_uri'].split('/', 2)[1].split('-', 2)
+        locality = None
+
+        try:
+            uri = FrbrUri.parse(self.kwargs['frbr_uri'])
+            country = uri.country
+            locality = uri.locality
+
+        except ValueError:
+            # not a valid URI; make some guesses
+            parts = self.kwargs['frbr_uri'].split('/')
+            place = parts[2] if parts[1] == 'akn' else parts[1]
+            if '-' in place:
+                country, locality = place.split('-')
+            else:
+                country = place
 
         # country
         try:
-            self.country = Country.for_code(parts[0])
+            self.country = Country.for_code(country)
```

Comment:
There's a helper method on Country that will take a place code and give you a country and locality all at once now.